app/services/publisher_promo_token2018q1_generator.rb

```diff
@@ -0,0 +1,15 @@
+# Generates a token for a publisher for 2018q1 promotion
+class PublisherPromoToken2018q1Generator < BaseService
```

Comment:
you can just call this PublisherPromoTokenGenerator

then have it take a param in `#initialize` for `promo_id:` and blow up if it's not the hardcoded code (free-bats-2018q1)